app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -171,4 +189,70 @@ private void deleteStudentImagesRecursive(StudentImage studentImage, String reas
         studentImageDao.delete(studentImage);
         Log.i(getClass().getName(), "StudentImage with the id " + studentImage.getId() + " has been deleted because " + reason);
     }
+
+    public synchronized void trainClassifier(){
+        // Initiate training if a StudentImageCollectionEvent has not been trained yet
+        if (studentImageCollectionEventDao.queryBuilder().where(StudentImageCollectionEventDao.Properties.SvmTrainingExecuted.eq(false)).count() > 0){
+            List<StudentImage> studentImages = studentImageDao.queryBuilder()
```

Comment:
@jogrimst Oh no. I just realized, that it's correct this way. For the training you will always need _all_ StudentImageFeatures. Here all the StudentImages with a StudentImageFeature are returned.